app/views/devise/registrations/new.html.slim

```diff
@@ -23,7 +23,11 @@ h2 = t('.title')
       .columns.small-4
         = f.input :birthdate, as: :string, input_html: {type: :date}
       .columns.small-6
-        = f.input :birthplace
+        .row
+          .columns.small-8
+            = f.input :birthplace
+          .columns.small-4
+            = f.input :birthplace_district, collection: YAML.load(File.read("app/services/districts.it.yml")).invert
```

Comment:
Non fare questo lavoro nella view. Non le compete.

Carica la collection nel controller in una variabile `@districts` e usala nella view.
A quel punto il controller non dovrebbe comunque essere incaricato di caricare il file yml. Il controller deve solo fare qualcosa tipo `@districts = District.all` ad esempio. 

Puoi avere un PORO (PlainOldRubyObject) dentro la cartella models `District` che  è una semplice classe con un metodo `all`. Se vuoi fare ulteriorimente il figo tale classe potrebbe essere un singleton cosicchè carichi il file una sola volta.

Fino a che controller e view si scambiano 3-5 oggetti per me va bene, anche se la teoria vorrebbe massimo uno (ma non creerei viewobject solo per questa variabile)